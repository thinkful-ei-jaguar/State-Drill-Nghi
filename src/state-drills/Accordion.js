import React from 'react';

class Accordion extends React.Component{
  // How data is passed via components
  static defaultProps = {
  	sections: []
  };
  // Use state to make data interative
  state = {
  	indexClicked: null
  };

  changeState = id => {
  	this.setState({
  	  indexClicked: id
  	});
  }
  
  // Cannot use arrow function because we want
  printSection(section, index, indexClicked) {
  	return (
  	  <li key={index}>
  		<button onClick={() => this.changeState(index)} >{section.title}
  	    </button>
  	    {indexClicked === index && <p>{section.content}</p>}
      </li>
  	);
  }

  render() {
  	return (
  	  <ul>
  	    // Printing new array of sections that also includes the element
  	    // Cannot print object if it's just this.props.sections
  	  	{this.props.sections.map((section, index) => 
  	  	this.printSection(section, index, this.state.indexClicked))}
  	  </ul>
  	);
  }


}

export default Accordion;

