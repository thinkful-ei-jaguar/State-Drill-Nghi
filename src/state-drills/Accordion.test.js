// Import libraries
import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Import module/component
import Accordion from './Accordion';

describe('Smoke test', () => {
  it('renders without error', () => {
  	// Creates div element
  	const div = document.createElement('div');
  	// Renders Component in div
  	ReactDom.render(<Accordion />, div);
  	// Cleanup div
  	ReactDom.unmountComponentAtNode(div);
  })}
)

describe('Snapshot tests', () => {
  const data = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
];

  it('renders when prop is not supplied', () => {
  	// Shallow creates an instance for us to interact with and creates simulations
  	const wrapper = shallow(<Accordion />);
  	// Create JSON version of the wrapper and compare the UI elements before and after
  	expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('renders no sections as active by default', () => {
  	const wrapper = shallow(<Accordion sections={data} />);
  	expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('opens clicked section', () => {
  	const wrapper = shallow(<Accordion sections={data} />);
  	// Find the first button in the wrapper and mock a click
  	wrapper.find('button').at(0).simulate('click');
  	expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('opens last section when multiple sections have been clicked', () => {
  	const wrapper = shallow(<Accordion sections={data} />);
  	wrapper.find('button').at(0).simulate('click');
  	wrapper.find('button').at(1).simulate('click');
  	wrapper.find('button').at(2).simulate('click');
  	expect(toJson(wrapper)).toMatchSnapshot();
  })
})





