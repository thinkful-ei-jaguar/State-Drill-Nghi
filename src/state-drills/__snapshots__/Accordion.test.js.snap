// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests opens clicked section 1`] = `
<ul>
  // Printing new array of sections that also includes the element // Cannot print object if it's just this.props.sections
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Snapshot tests opens last section when multiple sections have been clicked 1`] = `
<ul>
  // Printing new array of sections that also includes the element // Cannot print object if it's just this.props.sections
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Snapshot tests renders no sections as active by default 1`] = `
<ul>
  // Printing new array of sections that also includes the element // Cannot print object if it's just this.props.sections
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Snapshot tests renders when prop is not supplied 1`] = `
<ul>
  // Printing new array of sections that also includes the element // Cannot print object if it's just this.props.sections
</ul>
`;
